airflow/contrib/hooks/gcp_api_base_hook.py

```diff
@@ -159,6 +159,7 @@ def _get_field(self, f, default=None):
     def project_id(self):
         return self._get_field('project')
 
+    @staticmethod
```

Comment:
@mik-laj : I think this will not work for decorating class methods. I don't remember exactly why but I recall that you cannot use static methods decorated with @staticmethod to use as class method decorators. There is some pythonic subtlety here, as far as I remember it results in 'self' in the wrapper is wrong in this case. In order to make it works, you need to make them using older staticmethod() call. But I am curious to find out if you can do it with decorator.

Reply:
Why in that case do the tests pass correctly when it is wrong? I will return to it in my free time.